tRecorderApi/api/models/project.py

```diff
@@ -1,458 +1,453 @@
-    #     return data_dic
+        unique_together = (("version", "anthology", "language", "mode", "book"),)
+
+    # def __str__(self):
+    #     return '{}-{}-{} ({})'.format(self.language, self.version, self.book, self.id)
+
+    @staticmethod
+    def get_projects(language, book, anthology, version, published):
+        project_list = []
+        project_filter = {}
+
+        if language is not None:
+            project_filter["language__slug"] = language
+        if version is not None:
+            project_filter["version__slug"] = version
+        if book is not None:
+            project_filter["book__slug"] = book
+        if published is not None:
+            project_filter["published"] = published
+        if anthology is not None:
+            project_filter["anthology__slug"] = anthology
+
+        # filter projects based on the project being requested
+        projects = Project.objects.filter(**project_filter)
+
+        for project in projects:
+            dic = {"id": project.id,
+                   "published": project.published,
+                   "contributors": [],
+                   "version": {
+                       "slug": project.version.slug,
+                       "name": project.version.name
+                   },
+                   "anthology": {
+                       "slug": project.anthology.slug,
+                       "name": project.anthology.name
+                   },
+                   "language": {
+                       "slug": project.language.slug,
+                       "name": project.language.name
+                   },
+                   "book": {
+                       "slug": project.book.slug,
+                       "name": project.book.name,
+                       "number": project.book.number
+                   }
+                   }
+            latest_take = Take.objects.filter(chunk__chapter__project=project) \
+                .latest("date_modified")
+
+            dic["date_modified"] = latest_take.date_modified
+
+            min_check_level = Chapter.objects.all().values_list('checked_level') \
+                .order_by('checked_level')[0][0]
+
+            chunks_done = Chapter.objects.all().values_list('chunk').count()
+            takes = Take.objects.filter(chunk__chapter__project=project)
+
+            dic["checked_level"] = min_check_level
+            book_name_slug = project.book.slug
+            chunkInfo = []
+            # TODO 'os' will not be useful when the code migrate to AWS
+            for dirpath, dirnames, files in os.walk(os.path.abspath('static/chunks/')):
+                if dirpath[-3:] == book_name_slug:
+                    for fname in os.listdir(dirpath):
+                        f = open(os.path.join(dirpath, fname), "r")
+                        sus = json.loads(f.read())
+                        chunkInfo = sus
+                    break
+            total_chunk = float(len(chunkInfo))

```

Comment:
casting to a float is unnecessary, division by default is floating point division in python ( // for integer division)